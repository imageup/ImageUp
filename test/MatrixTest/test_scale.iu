function main() -> int
{
    matrix: arr(3,4) = [1,2,3,4|5,6,7,8|9,10,11,12];
    int : i;
    int : j;
    prints("matrix before");
    string : out;
    for(i = 0; i < RowLen(arr); i=i+1)
    {
        out = "row " + StrParse(i) + ": ";
        for(j = 0; j < ColLen(arr); j=j+1)
        {
            out = out + " " + StrParsef(arr[i][j]);
        }
        prints(out);
    }
    scale(arr, 2.4);
    prints("matrix after");
    for(i = 0; i < RowLen(arr); i=i+1)
    {
        out = "row " + StrParse(i) + ": ";
        for(j = 0; j < ColLen(arr); j=j+1)
        {
            out = out + " " + StrParsef(arr[i][j]);
        }
        prints(out);
    }
    return 0;
}