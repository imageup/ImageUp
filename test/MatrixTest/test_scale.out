matrix before
row 0:  1.00000 2.00000 3.00000 4.00000
row 1:  5.00000 6.00000 7.00000 8.00000
row 2:  9.00000 10.0000 11.0000 12.0000
matrix after
row 0:  2.40000 4.80000 7.20000 9.60000
row 1:  5.00000 6.00000 7.00000 8.00000
row 2:  9.00000 10.0000 11.0000 12.0000
